src/eosjs-serialize.ts

```diff
@@ -350,6 +350,10 @@ export class SerialBuffer { // tslint:disable-line max-classes-per-file
         if (typeof s !== 'string') {
             throw new Error('Expected string containing name');
         }
+        const regex = new RegExp(/^[.1-5a-z]{1,12}[1-5a-j]?(?<!\.)$/);
+        if (!regex.test(s)) {
+            throw new Error('Name should be less than 13 characters, not end with a period, and only contain the following symbols .12345abcdefghijklmnopqrstuvwxyz'); // tslint:disable-line
```

Comment:
Less than 13 or 12

May end with periods

Reply:
e.g. `"x"`, `"x."`, `"x.."` are all valid, but equivalent

Reply:
If all ending periods are removed, the best option in those examples is `"x"`. Is there a use-case to add periods although they will be removed by the serializing code?

If there isn't a use-case, it might be better to notify the users of this with either the current error or a warning. Otherwise the removal happens within the code with the possibility of the user being unaware.

Reply:
Since it's adding new restrictions to an existing library, there's a possibility it could break existing users. Disallowing something definitely incorrect, such as `"AbCdE"` should be ok.

Historical note: at one point eosjs produced `"............"` for empty names. To disallow it on input now seems risky.

Reply:
Adjusted the code to allow ending periods.  Also, changed the error message to give more info on the 13th character.